### PR TITLE
5211 program indicators

### LINKED ISSUE #5211: Indicators: Add program_indicator table
## Is your feature request related to a problem? Please describe 👀

Add the program_indicator table as defined here:

https://github.com/msupply-foundation/open-msupply/issues/5134

## Describe the solution you'd like 🎁

- [ ] add migration
- [ ] add repo boilerplate
- [ ] add service boilerplate
- [ ] add sync translator

### Moneyworks Jobcode 🧰

OMS:CIMDV

### PR DESCRIPTION
**Pull request by Chris-Petty**
Fixes #5211

# 👩🏻‍💻 What does this PR do?

- Migration for program_indicators table
- Repo layer
- sync translation
- Tests 

## 💌 Any notes for the reviewer?

Didn't do service layer let alone GQL layer. I don't think you'd generally query just for program_indicators? Maybe you'd add it to the `Requisition` struct...

https://github.com/msupply-foundation/open-msupply/blob/67af9f84ee3f655cf7cd33236fdc2c3e4c459985/server/repository/src/db_diesel/requisition/requisition.rs#L30-L46

Not for this PR though I think!

# 🧪 Testing

**DEV**
- [ ] OG central with some indicators configured
- [ ] Initialise your site to OMS (can be OMS central)
- [ ] See that the new table is created and populated

### CODE REVIEW

File: server/repository/src/db_diesel/program_indicator_row.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+use crate::{repository_error::RepositoryError, Upsert};
+
+use diesel::prelude::*;
+
+table! {
+    program_indicator (id) {
+        id -> Text,
+        program_id -> Text,
+        code -> Text,
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/program_indicator_row.rs:11]: In the migration you have code as Nullable

File: server/repository/src/db_diesel/program_indicator_row.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        is_active -> Bool,
+    }
+}
+
+#[derive(Clone, Insertable, Queryable, Debug, PartialEq, AsChangeset, Default)]
+#[diesel(table_name = program_indicator)]
+pub struct ProgramIndicatorRow {
+    pub id: String,
+    pub program_id: String,
+    pub code: String,
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/program_indicator_row.rs:21]: Same comment as above Optional code?

File: server/repository/src/migrations/v2_04_00/program_indicator_create_table.rs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    fn migrate(&self, connection: &StorageConnection) -> anyhow::Result<()> {
+        // Item variant Table
+        sql!(
+            connection,
+            r#"
+            CREATE TABLE program_indicator (
+                id TEXT PRIMARY KEY NOT NULL,
+                program_id TEXT NOT NULL REFERENCES program(id),
+                code TEXT,
+                is_active NOT NULL DEFAULT TRUE
```
Comment (roxy-dao) [inline server/repository/src/migrations/v2_04_00/program_indicator_create_table.rs:18]:
```suggestion
                is_active BOOL NOT NULL DEFAULT TRUE           
```

**Review by roxy-dao — CHANGES_REQUESTED** [review]
Yay thanks Chris!

Just a few comments below (:

File: server/repository/src/db_diesel/program_indicator_row.rs (hunk 1 above)
Comment (Chris-Petty) [reply server/repository/src/db_diesel/program_indicator_row.rs:11]: TRUE

**Review by roxy-dao — APPROVED** [review]
Brilliant~

**Pull request merged**